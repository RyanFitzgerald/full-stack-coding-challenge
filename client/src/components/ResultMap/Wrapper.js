import styled from 'styled-components';

const Wrapper = styled.div`
  border: 1px solid #e6e6e6;
  padding: 10px;
`;

export default Wrapper;