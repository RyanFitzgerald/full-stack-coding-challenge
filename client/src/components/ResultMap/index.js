import React from 'react';
import { Marker } from 'react-google-maps';
import { mapAPIKey } from '../../config';

import Map from '../Map';
import Wrapper from './Wrapper';

function renderMarker(address, key) {
  return (
    <Marker 
      key={key}
      title={address.formattedAddress}
      position={{lat: address.location.coordinates[1], lng: address.location.coordinates[0]}} 
    />
  );
}

function ResultMap({ data }) {
  return (
    <Wrapper>
      <Map
        googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${mapAPIKey}`}
        loadingElement={<div style={{ height: `100%` }} />}
        containerElement={<div style={{ height: `400px` }} />}
        mapElement={<div style={{ height: `100%` }} />}
      >
        {data.map((address, key) => renderMarker(address, key))}
      </Map>
    </Wrapper>
  );
}

export default ResultMap;