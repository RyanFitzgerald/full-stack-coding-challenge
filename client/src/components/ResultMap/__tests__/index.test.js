import React from 'react';

import ResultMap from '../index';

describe('<ResultMap />', () => {
  it('should match the snapshot', () => {
    const data = [{
      _id: '5b428620b4a1f026a4c6058b',
      city: 'Lenexa',
      country: 'United States',
      dateAdded: '2018-07-08T21:46:08.417Z',
      formattedAddress: '15481 W 110th St, Lenexa, KS 66219, USA',
      location: {
        coordinates: [-94.7655029, 38.928542],
        type: 'Point'
      },
      postalCode: '66219',
      state: 'Kansas',
      street: 'West 110th Street',
      streetNumber: '15481'
    }];

    const renderedComponent = renderer.create(
      <ResultMap data={data} />
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });
});