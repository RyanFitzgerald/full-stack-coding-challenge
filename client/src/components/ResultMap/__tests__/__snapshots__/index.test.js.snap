// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultMap /> should match the snapshot 1`] = `
.c0 {
  border: 1px solid #e6e6e6;
  padding: 10px;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</div>
`;
